pkg/controller/application/application_status.go

```diff
@@ -237,3 +278,191 @@ func (r *ReconcileApplication) objectsDeepEquals(oldStatus []sigappv1beta1.Objec
 	}
 	return false
 }
+
+// updateAppRelationships updates the configmap resources related to the Hybrid
+// App
+func (r *ReconcileApplication) updateAppRelationships(app *sigappv1beta1.Application) error {
+	// build the new configmap
+	relationshipsConfigmap, err := r.buildRelationshipsConfigmap(app)
+
+	// Update existing configmap or else create new
+	// Configmap will have same name and namespace as associated application
+	configmapKey := types.NamespacedName{
+		Name:      app.GetName(),
+		Namespace: app.GetNamespace(),
+	}
+	err = r.Get(context.TODO(), configmapKey, &corev1.ConfigMap{})
+	// Create the configmap if not existing
+	if err != nil {
+		if errors.IsNotFound(err) {
+			err = r.Create(context.TODO(), relationshipsConfigmap)
+
+			return err
+		}
+		return err
+	}
+
+	// Update existing configmap
+	err = r.Update(context.TODO(), relationshipsConfigmap)
+
+	return err
+}
+
+// buildRelationshipsConfigmap builds a configmap of resources related to the
+// app
+func (r *ReconcileApplication) buildRelationshipsConfigmap(app *sigappv1beta1.Application) (*corev1.ConfigMap, error) {
+
+	resources, err := r.fetchApplicationComponents(app)
+	if err != nil {
+		return nil, err
+	}
+
+	relationships := []Relationship{}
+	for _, resource := range resources {
+		// Check if resource exists and add to relationships
+		hdplKey := types.NamespacedName{
+			Name:      resource.GetName(),
+			Namespace: resource.GetNamespace(),
+		}
+		hdpl := hdplv1alpha1.Deployable{}
+		err = r.Get(context.TODO(), hdplKey, &hdpl)
+		if err != nil {
```

Comment:
we want to skip non-hdpl (IsNotFound error), but let's make sure other errors are handled